DevAcademy/Scenes/PlaceDetail/PlaceDetailViewModel.swift

```diff
@@ -2,24 +2,161 @@ import SwiftUI
+    // MARK: PROPERTIES
+    @EnvironmentObject private var placesObservableObject: PlacesObservableObject
     @State var isTappedFavorite = false
     
     var place: Place
     init(place: Place) {
         self.place = place
     }
     
+    
+    // MARK: PLACE MAIN INFORMATION
+    // IMAGE
+    var placeImage: String {
+        if imageIsFetch {
+            return place.attributes.imageURL!
+        }
+        return "Obrázek není k dispozici"
+    }
+    
+    var imageIsFetch: Bool {
+        if attributeIsFetch(place.attributes.imageURL) {
+            return true
+        }
+        return false
+    }
+    
+    // TITLE
     var placeName: String {
-        place.properties.nazev
+        place.attributes.title
+    }
+    
+    // KIND
+    var kindOfPlace: String {
+        place.attributes.kind.rawValue
+    }
+    
+    // STREET
+    var placeStreet: String {
+        if attributeIsFetch(place.attributes.street) {
+            return place.attributes.street!
+        }
+        return ""
+    }
+    
+    var placeStreetNo: String {
+        if attributeIsFetch(place.attributes.streetNo) {
+            return place.attributes.streetNo!
+        }
+        return ""
+    }
+    
+    
+    // MARK: PLACE LINKS
+    // WEB
+    var placeWeb: String {
+        if webIsAvailable {
+            return checkAndFixHasprefix(link: place.attributes.web!).filteringWhiteSpace()
+        }
+        return "-"
+    }
+    
+    var webIsAvailable: Bool {
+        if attributeIsFetch(place.attributes.web) {
+            return true
+        }
+        return false
+    }
+    
+    var webPlaceholder: String {
+        simpleStringUrl(from: placeWeb)
     }
     
-    var placeImage: URL {
-        place.properties.obrId1
+    // PHONE
+    var placePhone: String { // place phone return "123456789", "0", "-"
+        if attributeIsFetch(place.attributes.phone) {
+            return place.attributes.phone!.filteringWhiteSpace()
+        }
+        return "-"
+    }
+    
+    var phonePlaceholder: String {
+        formatPhoneNumber(placePhone)
+    }
+    
+    // EMAIL
+    var placeEmail: String {
+        if attributeIsFetch(place.attributes.email) {
+            return place.attributes.email!.filteringWhiteSpace()
+        }
+        return "-"
+    }
+    
+    // PROGRAMME
+    var placeProgramme: String {
+        checkAndFixHasprefix(link: place.attributes.programme!) // can force unwrapp - check value != nil in programmeIsAvailable
+    }
+    
+    var programmeIsAvailable: Bool {
+        if attributeIsFetch(place.attributes.programme) {
+            return true
+        }
+        return false
+    }
+    
+    
+    // MARK: FUNCTIONS
+    func attributeIsFetch(_ attribute: String?) -> Bool {
+        if attribute != nil {
+            return true
+        }
+        return false
     }
     
     func addFavorites() {
         isTappedFavorite.toggle()
     }
-}
+    
+    func checkAndFixHasprefix(link: String) -> String {
```

Comment:
metody a pomocne promenne, ktere pouzivas pouze uvnitr objektu, muzes oznacit jako `private`, zlepsuje to citelnost i cas kompilace. 😉